qcfractal/interface/collections/reaction_dataset.py

```diff
@@ -234,21 +328,121 @@ def visualize(self,
             The requested figure.
         """
 
-        query = {"method": method, "basis": basis, "keywords": keywords, "program": program, "stoich": stoich}
+        query = {"method": method, "basis": basis, "keywords": keywords, "program": program, "stoichiometry": stoich}
         query = {k: v for k, v in query.items() if v is not None}
 
         return self._visualize(metric, bench, query=query, groupby=groupby, return_figure=return_figure, kind=kind)
 
+    def get_molecules(self, subset: Optional[Union[str, Set[str]]] = None,
+                      stoich: Union[str, List[str]] = "default") -> Union[pd.DataFrame, 'Molecule']:
+        """Queries full Molecules from the database.
+
+        Parameters
+        ----------
+        subset : Optional[Union[str, Set[str]]], optional
+            The index subset to query on
+        stoich : Union[str, List[str]], optional
+            The stoichiometries to pull from, either a single or multiple stoichiometries
+
+        Returns
+        -------
+        Union[pd.DataFrame, 'Molecule']
+            Either a DataFrame of indexed Molecules or a single Molecule if a single subset string was provided.
+        """
+
+        self._check_client()
+        self._check_state()
+        self._validate_stoich(stoich)
+
+        indexer, names = self._molecule_indexer(stoich, subset)
+        df = self._get_molecules(indexer)
+        df.index = pd.MultiIndex.from_tuples(df.index, names=names)
+
+        if isinstance(subset, str) and isinstance(stoich, str):
```

Comment:
This gives incorrect output when subset is a string and stoich is a string which corresponds to more than one molecule. For example,
```
s22.get_molecules(subset='Water Dimer', stoich='default1')
```
gives only one of two molecules.